### PR TITLE
Add table aws_cloudfront_origin_access_identity. Closes #423

### LINKED ISSUE #423: Add table aws_cloudfront_origin_access_identity
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by visiit**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_cloudfront_origin_access_identity []

PRETEST: tests/aws_cloudfront_origin_access_identity

TEST: tests/aws_cloudfront_origin_access_identity
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_cloudfront_origin_access_identity.named_test_resource: Creating...
aws_cloudfront_origin_access_identity.named_test_resource: Creation complete after 3s [id=ETMDHHGQX0Q1J]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

caller_reference = terraform-20210527062305683200000001
etag = E3MLHJ2KA61SH1
resource_aka = arn:aws:cloudfront:986325076436:originaccessidentity/ETMDHHGQX0Q1J
resource_id = ETMDHHGQX0Q1J
resource_name = turbottest29171
s3_canonical_user_id = 24953303a79a60a2515f910f9eee62f81e531aca4227e29079e6ade56dfbabbbd75626f793e6fda5969d41d2aae22c30

Running SQL query: test-get-query.sql
[
  {
    "comment": "turbottest29171",
    "etag": "E3MLHJ2KA61SH1",
    "id": "ETMDHHGQX0Q1J",
    "s3_canonical_user_id": "24953303a79a60a2515f910f9eee62f81e531aca4227e29079e6ade56dfbabbbd75626f793e6fda5969d41d2aae22c30"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "arn": "arn:aws:cloudfront:986325076436:origin-access-identity/ETMDHHGQX0Q1J",
    "caller_reference": "terraform-20210527062305683200000001",
    "id": "ETMDHHGQX0Q1J"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:cloudfront:986325076436:origin-access-identity/ETMDHHGQX0Q1J",
    "comment": "turbottest29171",
    "etag": "E3MLHJ2KA61SH1",
    "id": "ETMDHHGQX0Q1J"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:cloudfront:986325076436:origin-access-identity/ETMDHHGQX0Q1J"
    ],
    "title": "ETMDHHGQX0Q1J"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudfront_origin_access_identity

TEARDOWN: tests/aws_cloudfront_origin_access_identity

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
- Basic Info

select
  id,
  arn,
  comment,
  s3_canonical_user_id,
  etag
from
  aws_cloudfront_origin_access_identity;

+---------------+--------------------------------------------------------------------+------------+--------------------------------------------------------------------------------------------------+----------------+
| id            | arn                                                                | comment    | s3_canonical_user_id                                                                             | etag          |
+---------------+--------------------------------------------------------------------+------------+--------------------------------------------------------------------------------------------------+----------------+
| E1UPUXHENNF8L | arn:aws:cloudfront:986325076436:origin-access-identity/E1UPUXHENNF8L | somecommet | 3514d72f466972629e9d52904c9810b6c250a9c47ab60d0c6e69e61e292e7b71628f53dbb724e23efe7ce0d9c5f5a453 | E2JL2WUWVKLQUI |
+---------------+--------------------------------------------------------------------+------------+--------------------------------------------------------------------------------------------------+----------------+

- List origin access identity with comment

select
  id,
  arn,
  comment,
  caller_reference
from
  aws_cloudfront_origin_access_identity
where
  comment <> '';

+---------------+--------------------------------------------------------------------+------------+--------------------------------------+
| id            | arn                                                                | comment    | caller_reference                     |
+---------------+--------------------------------------------------------------------+------------+--------------------------------------+
| E1UPUXHENNF8L | arn:aws:cloudfront:986325076436:origin-access-identity/E1UPUXHENNF8L | somecommet | ff7bf47a-a84a-4c83-8baa-d8f4d14c32f5 |
+---------------+--------------------------------------------------------------------+------------+--------------------------------------+
```
</details>

### CODE REVIEW

File: aws-test/tests/aws_cloudfront_origin_access_identity/variables.tf [hunk 1]
```diff
@@ -0,0 +1,75 @@
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "integration-tests"
```
Comment (khushboo9024) [inline aws-test/tests/aws_cloudfront_origin_access_identity/variables.tf:9]:
```suggestion
  default     = "default"
```

File: aws/table_aws_cloudfront_origin_access_identity.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+		Description: "AWS CloudFront Origin Access Identity",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("id"),
+			ShouldIgnoreError: isNotFoundError([]string{"NoSuchCloudFrontOriginAccessIdentity"}),
+			Hydrate:           getCloudFrontOriginAccessIdentity,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listCloudFrontOriginAccessIdentities,
+		},
+		Columns: awsRegionalColumns([]*plugin.Column{
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_origin_access_identity.go:27]: This is not a regional resource, I don’t  think we need to use `awsRegionalColumns`,  please check .

File: aws/table_aws_cloudfront_origin_access_identity.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+				Hydrate:     getCloudFrontOriginAccessIdentity,
+			},
+			{
+				Name:        "comment",
+				Description: "The comment for this origin access identity.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Comment", "CloudFrontOriginAccessIdentity.CloudFrontOriginAccessIdentityConfig.Comment"),
+			},
+			{
+				Name:        "e_tag",
```
Comment (Subhajit97) [inline aws/table_aws_cloudfront_origin_access_identity.go:61]:
```suggestion
				Name:        "etag",
```

File: aws-test/tint.js [hunk 4]
```diff
@@ -384,7 +384,7 @@ const _runGraphqlQuery = function (test, query) {
   return new Promise((resolve, reject) => {
     try {
       var queryTmp = _renderToTmp(test, query.query);
-      var variablesTmp = _renderToTmp(test, query.variables, {});
+      //var variablesTmp = _renderToTmp(test, query.variables, {});
```
Comment (khushboo9024) [inline aws-test/tint.js:387]: Why we are commenting this ?

File: aws/table_aws_cloudfront_origin_access_identity.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+	plugin.Logger(ctx).Trace("getCloudFrontOriginAccessIdentityARN")
+	originAccessIdentityData := *originAccessIdentityID(h.Item)
+
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	commonColumnData := c.(*awsCommonColumnData)
+	arn := "arn:" + commonColumnData.Partition + ":cloudfront:" + commonColumnData.AccountId + ":originaccessidentity/" + originAccessIdentityData
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_origin_access_identity.go:151]: arn format should be `arn:${Partition}:cloudfront::${Account}:origin-access-identity/${Id}`

File: aws/table_aws_cloudfront_origin_access_identity.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+	plugin.Logger(ctx).Trace("getCloudFrontOriginAccessIdentityARN")
+	originAccessIdentityData := *originAccessIdentityID(h.Item)
+
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+
+	commonColumnData := c.(*awsCommonColumnData)
+	arn := "arn:" + commonColumnData.Partition + ":cloudfront:" + commonColumnData.AccountId + ":origin-access-identity/" + originAccessIdentityData
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_origin_access_identity.go:151]:
```suggestion
	arn := "arn:" + commonColumnData.Partition + ":cloudfront::" + commonColumnData.AccountId + ":origin-access-identity/" + originAccessIdentityData
```

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by khushboo9024 — APPROVED** [review]

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**